docs/concepts-throttling.md

```diff
@@ -0,0 +1,167 @@
+
+1. **The broker may read more entries or bytes from the bookies than the throttling limit.**
+
+   a) **The byte size of messages delivered to the client may exceed the configured threshold.**
+   
+     When you set the dispatch rate limit in bytes/throttling-period (`dispatchThrottlingRateInByte`/`ratePeriodInSecond`), the broker calculates $$the \ number \ of \ entries \ to \ read \ from \ bookies$$ in one throttling period through the following equation:
+   
+     $$
+     The \ number \ of \ entries \ to \ read \ from \ bookies = {{The \ total \ byte \ size \ to \ read} \over{The \ average \ byte \ size \ per \ entry}}
+     $$
+
+     By controlling $$the \ number \ of \ entries \ to \ read \ from \ bookies$$, the broker attempts to limit `the total byte size to read` below `the dispatch rate` within each throttling period. It reads messages from the bookies in the unit of `entry` and approximates the bytes of the next entry to read because it does not know the exact byte size of each entry before reading it.
+
+     The broker uses the following two metrics to get the average byte size per entry:
+
+      * Average publish size (`brk_ml_EntrySizeBuckets`): the average byte size per entry stored in the bookies when the broker receives a publish request.
+
+      * Average dispatch size (`entriesReadSize`/`entriesReadCount`): the average byte size per entry read from bookies, that is, the average byte size per entry sent to the client.
+      
+     The broker uses the average publish size in preference to the average dispatch size. If the average publish size is unavailable, then it uses the average dispatch size. When none of the two metrics are available, the broker only reads one entry at the first attempt.
+
+   **b) The number of messages delivered to the client may exceed the configured threshold.**
+     
+     When you set the dispatch rate limit in message-count/throttling-period (`dispatchThrottlingRateInMsg`/`ratePeriodInSecond`) and batching (`batch-send`) is enabled, the broker counts an entry as one message (despite the message count per entry) and calculates $$the \ number \ of \ entries \ to \ read \ from \ bookies$$ through the following equation:
+      
+     $$
+     The \ number \ of \ entries \ to \ read \ from \ bookies = {{The \ total \ number \ of \ messages \ to \ read} \over{The \ average \ message \ count \ per \ entry} \ (=1)}
+     $$
+
+     Since there is a number of messages per entry, the number of messages delivered to the client always exceeds or equals the configured threshold.
+
+   **Workaround**
+
+   Configuring `preciseDispatcherFlowControl` or `dispatchThrottlingOnBatchMessageEnabled` can mitigate the over-delivery issue. For example, turning on `preciseDispatcherFlowControl` can mitigate the limitation by pre-decrementing the quota using the approximated average message count per entry. See [Throttling configurations](#throttling-configurations) for more details.
```

Comment:
If this mitigation also helps the second limitation, please move this to the bottom of this section and mention these configs can help both limitations.

Reply:
@poorbarcode can this workaround resolve the limitation caused by the second case? I thought the over-delivery limitation can be caused by two cases, and this workaround only serves the first one. Can you please clarify this?

Reply:
Hi @momo-jun 
> can this workaround resolve the limitation caused by the second case? I thought the over-delivery limitation can be caused by two cases

Yes, you are right. Thanks for @heesung-sn